src/printer.js

```diff
@@ -562,7 +562,20 @@ function genericPrintNoParens(path, options, print) {
       parts.push("return");
 
       if (n.argument) {
-        parts.push(" ", path.call(print, "argument"));
+        if (n.argument.comments) {
+          parts.push(
+            n.loc.start.line === n.loc.end.line
```

Comment:
@jlongster Thanks for the feedback! Indeed, if my solution is too aggressive and not matching your views, maybe it's better to simply close this PR 👍.